x/mongo/driver/operation.go

```diff
@@ -1283,7 +1284,7 @@ func (op Operation) secondaryOK(desc description.SelectedServer) wiremessage.Que
 }
 
 func (Operation) canCompress(cmd string) bool {
-	if cmd == "isMaster" || cmd == "saslStart" || cmd == "saslContinue" || cmd == "getnonce" || cmd == "authenticate" ||
+	if cmd == internal.LegacyHello || cmd == "saslStart" || cmd == "saslContinue" || cmd == "getnonce" || cmd == "authenticate" ||
```

Comment:
This may additionally need "hello" as well based on: https://github.com/mongodb/specifications/blob/3da2c7816f6cdb449bd22603523c2dfc4b0af3e9/source/compression/OP_COMPRESSED.rst#messages-not-allowed-to-be-compressed

Reply:
Great catch. Added.